release.py

```diff
@@ -0,0 +1,192 @@
+    shell('git pull --ff-only')
+
+    # find latest release
+    latest_release = shell('git describe --tags --abbrev=0', capture_output=True)
+    latest_release_commit = shell('git rev-list -n 1 {}'.format(latest_release),
+        capture_output=True)
+
+    print('Latest version: {} ({})'.format(latest_release, latest_release_commit[:7]))
+    print('New version: {}'.format(new_release))
+
+    # check that release numbers are increasing
+    try:
+        if StrictVersion(latest_release[1:]) >= StrictVersion(new_release[1:]):
+            abort('Version numbers must be increasing')
+    except ValueError as e:
+        abort(str(e))
+
+    # edit changelog message
+    log = shell('git log --pretty=format:"- %s" {}..HEAD'.format(latest_release),
+        capture_output=True)
+    changelog = edit('\n'.join([
+        'Changelog',
+        log,
+        '',
+        '# Please enter a changelog since the latest release. Lines starting',
+        "# with '#' will be ignored, and an empty message aborts the release.",
+    ]))
+    log_lines = [line for line in changelog.splitlines() if line[:1] != '#']
+    changelog = '\n'.join(log_lines).strip()
+    if not changelog:
+        abort('Empty changelog, aborting')
+
+    # edit client/__init__.py and commit
+    init_file = 'client/__init__.py'
+    with open(init_file, 'r', encoding='utf-8') as f:
+        old_init = f.read()
+    new_init = re.sub(
+        r"^__version__ = '([a-zA-Z0-9.]+)'",
+        "__version__ = '{}'".format(new_release),
+        old_init)
+    if old_init != new_init:
+        print('Editing version string in {}'.format(init_file))
+        with open(init_file, 'w', encoding='utf-8') as f:
+            init = f.write(new_init)
+        shell('git add {}'.format(init_file))
+        shell('git commit -m "Bump version to {}"'.format(new_release))
+        shell('git push')
+
+    print('Uploading release to GitHub...')
+    shell('ok-publish')
+    github_release = post_request(
+        'https://api.github.com/repos/{}/releases'.format(GITHUB_REPO),
+        headers={
+            'Authorization': 'token ' + github_token,
+        },
+        json={
+            'tag_name': new_release,
+            'target_commitish': 'master',
+            'name': new_release,
+            'body': changelog,
+            'draft': False,
+            'prerelease': False
+        },
+    )
+    with open('ok', 'rb') as f:
+        github_asset = post_request(
+            'https://uploads.github.com/repos/{}/releases/{}/assets'.format(
+                GITHUB_REPO, github_release['id']),
+            params={
+                'name': 'ok'
+            },
+            headers={
+                'Authorization': 'token ' + github_token,
+                'Content-Type': 'application/octet-stream',
+            },
+            data=f,
+        )
+
+    print('Updating version on {}...'.format(OK_SERVER_URL))
+    # Create a fake assignment to log in. I'm not happy about this
```

Comment:
Out of scope for this PR: Getting a token seems like the one thing we should be able to support without an assignment. Maybe we should add a method somewhere that accepts all of the default values. 

I've opened up a URL in other scripts - but I don't recommend it for this file. 
```
def get_token():
    print("Get an access token for your OK account & paste it below")
    webbrowser.open('https://ok-oauth.app.cs61a.org/')
    return input("Your OK Token: ").strip()
```